qttypes/src/gui/qcolor.rs

```diff
@@ -0,0 +1,871 @@
+            return self->green();
+        })
+    }
+
+    pub fn green_f(&self) -> qreal {
+        cpp!(unsafe [self as "const QColor*"] -> qreal as "qreal" {
+            return self->greenF();
+        })
+    }
+
+    pub fn hsl_hue(&self) -> i32 {
+        cpp!(unsafe [self as "const QColor*"] -> i32 as "int" {
+            return self->hslHue();
+        })
+    }
+    pub fn hsl_hue_f(&self) -> qreal {
+        cpp!(unsafe [self as "const QColor*"] -> qreal as "qreal" {
+            return self->hslHueF();
+        })
+    }
+    pub fn hsl_saturation(&self) -> i32 {
+        cpp!(unsafe [self as "const QColor*"] -> i32 as "int" {
+            return self->hslSaturation();
+        })
+    }
+    pub fn hsl_saturation_f(&self) -> qreal {
+        cpp!(unsafe [self as "const QColor*"] -> qreal as "qreal" {
+            return self->hslSaturationF();
+        })
+    }
+    pub fn hsv_hue(&self) -> i32 {
+        cpp!(unsafe [self as "const QColor*"] -> i32 as "int" {
+            return self->hsvHue();
+        })
+    }
+    pub fn hsv_hue_f(&self) -> qreal {
+        cpp!(unsafe [self as "const QColor*"] -> qreal as "qreal" {
+            return self->hsvHueF();
+        })
+    }
+
+    pub fn hsv_saturation(&self) -> i32 {
+        cpp!(unsafe [self as "const QColor*"] -> i32 as "int" {
+            return self->hsvSaturation();
+        })
+    }
+
+    pub fn hsv_saturation_f(&self) -> qreal {
+        cpp!(unsafe [self as "const QColor*"] -> qreal as "qreal" {
+            return self->hsvSaturationF();
+        })
+    }
+
+    pub fn is_valid(&self) -> bool {
+        cpp!(unsafe [self as "const QColor*"] -> bool as "bool" {
+            return self->isValid();
+        })
+    }
+
+    pub fn lighter(&self, factor: Option<i32>) -> QColor {
+        let factor = match factor {
+            Some(factor) => factor,
+            None => 150,
+        };
+
+        cpp!(unsafe [self as "const QColor*", factor as "int"] -> QColor as "QColor" {
+            return self->lighter(factor);
+        })
+    }
+
+    pub fn lightness(&self) -> i32 {
+        cpp!(unsafe [self as "const QColor*"] -> i32 as "int" {
+            return self->lightness();
+        })
+    }
+
+    pub fn lightness_f(&self) -> qreal {
+        cpp!(unsafe [self as "const QColor*"] -> qreal as "qreal" {
+            return self->lightnessF();
+        })
+    }
+
+    pub fn magenta(&self) -> i32 {
+        cpp!(unsafe [self as "const QColor*"] -> i32 as "int" {
+            return self->magenta();
+        })
+    }
+
+    pub fn magenta_f(&self) -> qreal {
+        cpp!(unsafe [self as "const QColor*"] -> qreal as "qreal" {
+            return self->magentaF();
+        })
+    }
+
+    pub fn name(&self) -> QString {
+        cpp!(unsafe [self as "const QColor*"] -> QString as "QString" {
+            return self->name();
+        })
+    }
+
+    pub fn name_format(&self, format: NameFormat) -> QString {
```

Comment:
maybe this should be renamed `name_with_format`  or just have `name` that always take a format.
another option would be to just have a `name_argb` and `name_rgb`.